src/utils.jl

```diff
@@ -1,18 +1,13 @@
-function wdot{T}(w::Vector{T}, x::Vector{T}, y::Vector{T})
+function wdot{T}(wx::AbstractVector{T}, x::AbstractVector{T},
+                 wy::AbstractVector{T}, y::AbstractVector{T})
     out = zero(T)
     @inbounds @simd for i in 1:length(x)
```

Comment:
I guess it won't really ever be, but if these aren't `Array`s, then it would be safer to go `for i in eachindex(wx, x, wy, y)`, but...